rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -1513,7 +1750,18 @@ def batch_loss(
     ) -> tf.Tensor:
         tf_batch_data = self.batch_to_model_data_format(batch_in, self.data_signature)
 
-        sequence_lengths = self._get_sequence_lengths(tf_batch_data[TEXT_SEQ_LENGTH][0])
+        mask_sequence_text = self._get_mask_for(tf_batch_data, TEXT_SEQUENCE_LENGTH)
+        mask_sentence_text = self._get_mask_for(tf_batch_data, TEXT_SENTENCE_LENGTH)
+
+        if TEXT_SEQUENCE_LENGTH in tf_batch_data:
+            sequence_lengths = self._get_sequence_lengths(
+                tf_batch_data[TEXT_SEQUENCE_LENGTH][0]
+            )
+            sequence_lengths += 1  # add cls token
```

Comment:
I think it should be smth like
```suggestion
            sequence_lengths += 1  # add sentence features
```